cli/src/main/java/com/devonfw/tools/ide/common/SystemPath.java

```diff
@@ -167,11 +168,12 @@ public Path getPath(String tool) {
 
   /**
    * @param tool the name of the tool.
-   * @param path the new {@link #getPath(String) tool bin path}.
+   * @param path the new {@link #getPath(String) tool bin path} and {@link #paths}.
    */
   public void setPath(String tool, Path path) {
 
     this.tool2pathMap.put(tool, path);
+    this.paths.add(0, path);
```

Comment:
Salim did almost the same "fix" here:
https://github.com/devonfw/IDEasy/pull/143/files#diff-594121d7c6efe86c0927ed36bc6e69da51a4ee92f8ed2ac0958b118e3fc3f3ff

I will get in touch to discuss how we can get to the best solution.

Reply:
@hohwille I changed it to iterate also over the tool2pathMap as discussed.